nifi-api/src/main/java/org/apache/nifi/processor/ProcessContext.java

```diff
@@ -73,6 +73,15 @@ public interface ProcessContext extends PropertyContext, ClusterContext {
      */
     void yield();
 
+    /**
+     * <p>
+     * An alternative name for the yield() method so that it can be called from Python.
+     * </p>
+     */
+    default void yieldResources() {
```

Comment:
Changing the nifi-api is not in scope for this JIRA or PR.  Other solutions should be identified.

Reply:
To me it seems like a straight forward _extension_ of the api, which provides an alternative name for a given function without breaking anything. What better alternatives do you see?

Reply:
Introducing an alias only required by the Python side into the Java codebase seems not a good enough reason to me personally. 
The surface of such a core API should be as narrow as possible I think. 

From what I've gather from other issues, the mailing list and Slack is that the Python API is intentionally kept simplistic. I'm not certain the Python API should allow yielding explicitly at all. 

I personally thought the original commit was a better approach by making the yield implicit and thus enforcing a good practice used widely by `ListX` style Java processors.

Reply:
It does not extend the API.  It creates a method with a slightly different name then calls the other.

I am not presenting alternatives but seeking alternatives.  My role in this is as a reviewer considering whether this change is warranted and my assertion at this point is that we should seek alternatives.

Reply:
I agree with the approach of keeping the python API simple, on the other hand, we are in the perfect time to modify/extend it based on our experience with it. It's not final in M2, but a very cool thing to have, so people started using it and naturally hit limitations. Before going GA we have the chance to adjust it to our needs, later on our hands will be much more tied because of the compatibility we have to respect. 
The "Handling of the contents of the failure result in the python FlowFileTransform API"  thread on the dev mailing list was I think one of the first occasions where the limits of the current API was hit and people don't completely agree on the general approach moving forward. From this point most of the questions become a tradeoff:
-do we want to allow the users to implement python controller services for eg?
-how much feature-parity we are going to provide with the java api when creating processors?
-how simple we would like to keep the python api?
-how "intrusive" we let python related code changes to be on the java side?

I feel like we all have different answers to these questions above. 
My personal preference is the feature parity way, but I'm pretty sure there is no consensus on that.

This is way off the scope of this PR, but I think we should find a general approach and common goals for the python API and make sure these are clear for everyone and kept in my before submitting changes, otherwise we will end up discussing things on PRs that are more like principles and not implementation details.

Reply:
Using py4j [get_method](https://www.py4j.org/py4j_java_gateway.html#py4j.java_gateway.get_method) might allow us to retrieve a method reference to yield without the keyword conflict, since it takes the method name as a string.

Reply:
If you are proposing that there are more general/fundamental things to nail down prior to the 2.0 release as it relates to the Python API please initiate a discussion thread or wiki with a proposal.

For the specific issue I flagged with this PR we need other options that do not leak such a concern into the nifi-api.

Reply:
I would agree that we should avoid changing the nifi-api for this. I think the suggestion provided by @szaszm is very reasonable. The Python side already provides a wrapper around the ProcessContext proxy, so we can certainly add a `yield_resources` method to that Python-only ProcessContext, and have it use  `get_method`. 

Alternatively, we could take the approach of updating the return Object (`FlowFileSourceResult`) to provide an `isYielded()` method, and have the Python side always return an object, rather than `null`, if `yieldResources` were called.

That said, I do think that @szaszm's solution of just retrieving the method by name and invoking it directly is a better solution. Just thinking through alternatives.